vspace/src/main/java/edu/asu/diging/vspace/web/staff/DashboardController.java

```diff
@@ -2,32 +2,67 @@
 
 import java.util.List;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
 import edu.asu.diging.vspace.core.data.ModuleRepository;
 import edu.asu.diging.vspace.core.data.SpaceRepository;
+import edu.asu.diging.vspace.core.model.ISpaceLink;
 import edu.asu.diging.vspace.core.model.impl.Module;
 import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.services.ISpaceLinkManager;
 
 @Controller
 public class DashboardController {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
 
     @Autowired
     private SpaceRepository spaceRepo;
 
     @Autowired
     private ModuleRepository moduleRepo;
+    
+    @Autowired
+    private ISpaceLinkManager spaceLinkManager;
 
     @RequestMapping("/staff/dashboard")
     public String displayDashboard(Model model) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Integer spaceLinksSize=0;
+        if(authentication.getName().equals("admin")) {
+            List<ISpaceLink> spaceLinks = spaceLinkManager.findSpaceLinksWithSourceNull();
+            spaceLinksSize = spaceLinks.size();
```

Comment:
don't load all the links into memory just to find out how many there are, create a repository method for this.

Reply:
changed